### ISSUE TITLE
Twitter preview cut off

### ISSUE THREAD

**Issue by noiob** [telegramdesktop/tdesktop#2496]
The twitter preview on Telegram desktop is always cut off, even if it doesn't have to be (try tweets with multiple lines). The app doesn't do it and even if there are a few line breaks a tweet with an image is probably still even larger. It's just an annoyance.

Edit: I don't know if this should be a separate issue, but it'd also be nice if the preview didn't pick the very last image from a tweet with multiple images, since that one is probably the punch line or something

**Comment by stek29** [comment]
@noiob what about other apps (Android, iOS, Webogram...)

**Comment by noiob** [comment]
@stek29 Android and iOS don't cut off the preview. Webogram doesn't either. You can check it yourself, here's an example tweet: https://twitter.com/conradhackett/status/783029175791808512

**Comment by erisdev** [comment]
this is extremely annoying. tweets are guaranteed max 140 characters so imo they should never be cut off in the first place.

**Comment by nachoparker** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  pkbrister@zoho.com
    Domain ownyourbits.com has exceeded the max emails per hour (3/3 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext1.iad.github.net ([192.30.252.192]:47783 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by skywalker.onesite.co with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1cVlKF-003AXF-V8
	for nacho@ownyourbits.com; Mon, 23 Jan 2017 15:34:52 -0500
Date: Mon, 23 Jan 2017 12:34:05 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1485203645;
	bh=qbKh6osmuuSQArjrA9xRvq1dI7AK643Kokk0FflU0DA=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=ytuOrwRft9nJhfY6nKoLeNV1Ii9o2hj6kkhFPpXxxWZXOaT3n3og9Z0VyNccngPwp
	 gpgnH8Y8idvHzxG0XxiEPpuQJiqyfAdpRHVpIsEpk5oSWbaomV8JLsIpPXjLKkvrcO
	 1yPBcBK8or3BpDP4z2wb518Z+o6bpblnJkeX3tD8=

**Comment by tdesktop-closer** [comment]
Hey there!

We're automatically closing this issue since there was no activity in this issue since 493 days ago. We therefore assume that the user has lost interest or resolved the problem on their own. Closed issues that remain inactive for a long period may get automatically locked.

Don't worry though; if this is in error, let us know with a comment and we'll be happy to reopen the issue.

Thanks!

<sub>(Please note that this is an [automated](https://github.com/auchri/github_auto_closer) comment.)</sub>

**Comment by Remu-rin** [comment]
This is still an issue even for text-only tweets in single line. Seems like tdesktop wasn't updated for 280 characters in tweets, while Android client can show all 280 in preview.

**Comment by stek29** [comment]
Something tells me it's a feature, but I don't understand its purpose.
I've made a sample link: https://stek29.rocks/test_tdesktop_preview

Here's what TDesktop shows:
![2018-06-06 12 49 19](https://user-images.githubusercontent.com/11808223/41030909-10aef39e-6988-11e8-813a-588bf4a58f92.jpg)

And here's what other clients (macOS native app in this case) show:
<img width="298" alt="screen shot 2018-06-06 at 12 49 51" src="https://user-images.githubusercontent.com/11808223/41030938-2163c7c8-6988-11e8-95c0-3eec289bcb72.png">

As you can see, server already does turn newlines into spaces when there are more than 9 lines, but TDesktop cuts preview on third line.

**Comment by stek29** [comment]
(cc @john-preston)

**Comment by john-preston** [comment]
@stek29 It's not a feature, but an absence of one. My Text renderer doesn't allow to place a float picture and let text float around the picture (take less space, while the picture is at the right, and after that start taking all the width.

So the simple solution was to avoid situations (in link previews) where there could be an image at the right and text is still going after the image for many lines. So the text is just limited to the possible image height. 🙈

**Comment by stek29** [comment]
@john-preston What if text just stays narrow and picture just stays on left?

**Comment by john-preston** [comment]
@stek29 You mean on the right?

**Comment by stek29** [comment]
@john-preston Yep, on the right, was a typo.

**Comment by john-preston** [comment]
@stek29 Not cool :( but perhaps it should be done that way for now..

**Comment by atomicthumbs** [comment]
Seconding the issue with it truncating tweets for no good reason after the Twitter 280 character change. Seems to cut them off at 175 characters, which is pretty arbitrary!

**Comment by john-preston** [comment]
@atomicthumbs I think it cuts them at four lines of text.

**Comment by salnkv** [comment]
I there any way to avoid cut off of 280 char tweets?

**Comment by sunscreem** [comment]
Adding the voices looking for 280 char limit and new line support,.

**Comment by atomicthumbs** [comment]
it has been four years and telegram will still happily cut the last four characters off a tweet in the middle of a word

**Comment by atomicthumbs** [comment]
Nope. Still an issue. Just because y'all haven't fixed the bug yet doesn't mean it's no longer a bug.

**Comment by Remu-rin** [comment]
Still relevant.

**Comment by noiob** [comment]
This issue has been open for over six years and is still extremely annoying, especially with services like fxtwitter putting quote tweets in the preview. In other news, if it gets ignored for another year or so maybe Twitter will have stopped being relevant

**Comment by enchained** [comment]
> @stek29 It's not a feature, but an absence of one. My Text renderer doesn't allow to place a float picture and let text float around the picture (take less space, while the picture is at the right, and after that start taking all the width.
> 
> So the simple solution was to avoid situations (in link previews) where there could be an image at the right and text is still going after the image for many lines. So the text is just limited to the possible image height. 🙈

@john-preston Any chance for an update on this matter? In my example https://github.com/telegramdesktop/tdesktop/issues/27186 it seems like cut off at 4 lines happens even if the image is not floating to the right and displayed larger at the bottom instead. 

Since the server now allows to pick image position for the preview, can this 4 line cut off be disabled at least for the previews with the image at the bottom?
